CMakeLists.txt

```diff
@@ -40,7 +40,11 @@ set_property(   GLOBAL PROPERTY PROPERTY_OUTPUT_FORMAT elf32-little${ARCH})
 # compiler options needed by all source files. All zephyr libraries,
 # including the library named "zephyr" link with this library to
 # obtain these flags.
+# Linker flags will be propagated to executable targets which links
+# to zephyr_interface, either directly or through other libraries which
+# depends upon zephyr_interface, e.g. zephyr itself.
 add_library(zephyr_interface INTERFACE)
+target_link_libraries(zephyr_interface INTERFACE ${TOOLCHAIN_LIBS})
```

Comment:
The zephyr_link_libraries() API should be used when a linker flag needs to be added to all Zephyr sources.

so

zephyr_link_libraries(${TOOLCHAIN_LIBS})